src/rest_api/classes/construct/widgets/isolation.clj

```diff
@@ -0,0 +1,34 @@
+(ns rest-api.classes.construct.widgets.isolation
+  (:require
+   [rest-api.classes.generic-fields :as generic]
+   [rest-api.formatters.object :as obj :refer [pack-obj]]))
+
+(defn clone [c]
+  {:data (some->> (:construct/clone c)
+                  (map pack-obj)
+                  (first))
+   :description "the clone of this construct"})
+
+(defn construction-summary [c]
+  {:data (first (:construct/construction-summary c))
+   :d (:db/id c)
+   :description "Construction details for the transgene"})
+
+(defn historical-genes [c]
+  {:data (some->> (:construct/historical-gene c)
+                  (map (fn [h]
+                         (let [gene (:construct.historical-gene/gene h)]
+                           {:text (pack-obj gene)
```

Comment:
Best to call `(pack-obj gene)` once, assign to a var in the let expression above

Reply:
fixed with next commit